sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -619,6 +620,104 @@ case class DescribeTableCommand(
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   DESCRIBE [EXTENDED|FORMATTED] table_name column_name;
+ * }}}
+ */
+case class DescribeColumnCommand(
+    table: TableIdentifier,
+    column: String,
+    isFormatted: Boolean)
+  extends RunnableCommand {
+
+  override val output: Seq[Attribute] = {
+    // The displayed names are based on Hive.
+    // (Link for the corresponding Hive Jira: https://issues.apache.org/jira/browse/HIVE-7050)
+    if (isFormatted) {
+      Seq(
+        AttributeReference("col_name", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "name of the column").build())(),
+        AttributeReference("data_type", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "data type of the column").build())(),
+        AttributeReference("min", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "min value of the column").build())(),
+        AttributeReference("max", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "max value of the column").build())(),
+        AttributeReference("num_nulls", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "number of nulls of the column").build())(),
+        AttributeReference("distinct_count", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "distinct count of the column").build())(),
+        AttributeReference("avg_col_len", StringType, nullable = true,
+          new MetadataBuilder().putString("comment",
+            "average length of the values of the column").build())(),
+        AttributeReference("max_col_len", StringType, nullable = true,
+          new MetadataBuilder().putString("comment",
+            "maximum length of the values of the column").build())(),
+        AttributeReference("comment", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "comment of the column").build())())
+    } else {
+      Seq(
+        AttributeReference("col_name", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "name of the column").build())(),
+        AttributeReference("data_type", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "data type of the column").build())(),
+        AttributeReference("comment", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "comment of the column").build())())
+    }
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val resolver = sparkSession.sessionState.conf.resolver
+    val catalogTable = catalog.getTempViewOrPermanentTableMetadata(table)
+    val attribute = {
+      val field = catalogTable.schema.find(f => resolver(f.name, column))
+      field.getOrElse {
+        if (column.contains(".")) {
```

Comment:
what if users run
```
DESC t1 `col1.a`
```
They are looking for a special column named "col1.a", but not nested column